server/src/main/java/org/candlepin/model/Pool.java

```diff
@@ -544,107 +593,186 @@ public void setOrderNumber(String orderNumber) {
         this.orderNumber = orderNumber;
     }
 
-    public boolean hasAttribute(String key) {
-        return this.findAttribute(key) != null;
+    /**
+     * Retrieves the attributes for this pool. If this pool does not have any attributes,
+     * this method returns an empty map.
+     *
+     * @return
+     *  a map containing the attributes for this pool
+     */
+    public Map<String, String> getAttributes() {
+        return Collections.unmodifiableMap(this.attributes);
     }
 
     /**
-     * Attribute comparison helper, safe to use even if property is null.
+     * Retrieves the value associated with the given attribute. If the attribute is not set, this
+     * method returns null.
+     *
+     * @param key
+     *  The key (name) of the attribute to lookup
      *
-     * Used primarily in the javascript rules.
+     * @throws IllegalArgumentException
+     *  if key is null
      *
-     * @param key Desired attribute.
-     * @param expectedValue Expected value.
-     * @return true if the pool has the given attribute and it is equal to the value,
-     * false otherwise.
+     * @return
+     *  the value set for the given attribute, or null if the attribute is not set
      */
-    public boolean attributeEquals(String key, String expectedValue) {
-        String val = getAttributeValue(key);
-        if (val != null && val.equals(expectedValue))  {
-            return true;
+    @XmlTransient
+    public String getAttribute(String key) {
```

Comment:
Agreed.  It pains me to see things so wordy, but given Attributes have both a name and value, it's probably the right call.  Unless of course `getAttribute` returned an Attribute object which I don't think it really should.